<!--REQUEST_ID=16c359a8-5562-4a19-8aad-f41c96c258e9-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46033&attempt_id=ad8dee5b-1bf4-4d83-a08c-f9192baf1cdc)

Attempted: monero, monero-gui

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/bin/moc -D_THREAD_SAFE -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/mkspecs/macx-clang -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/monero/include -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/src/libwalletqt -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/src/QR-Code-generator -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/src -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/monero/src -I/nix/store/p4cg75k1k5x6xaqwbcy6llk6c6fffqs4-pcsclite-1.8.23-dev/include/PCSC -I/nix/store/9nprgv6j52mkc0p5brk5sq670xsgdy5w-qtdeclarative-5.11.1-dev/include/QtQuick -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtWidgets -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtGui -I/nix/store/9nprgv6j52mkc0p5brk5sq670xsgdy5w-qtdeclarative-5.11.1-dev/include/QtQml -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtNetwork -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtCore -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include -I/nix/store/1jn4cdi9xggshrqvfwmxjhhhxihg2pkb-libxml2-2.9.8-dev/include -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/7677nf05z1r9nxvzygi1apcr9hwisdws-libxslt-1.1.32-dev/include -I/nix/store/8zr244xgnxvpkp0mzcyzf60fbh6afqsl-openssl-1.0.2p-dev/include -I/nix/store/kik9k5xd18ypqmz4gp9sblhfszsbva56-sqlite-3.24.0-dev/include -I/nix/store/jbf08c95535nam27myywqcx8md7vxhl4-harfbuzz-icu-1.8.8-dev/include -I/nix/store/mhaq82i9s2v84jyp5mzkdrgrws39fyz9-graphite2-1.3.6/include -I/nix/store/gx085743laxybi80pjbggaq3lp8385vy-icu4c-59.1-dev/include -I/nix/store/g8bgy3i1423ffz65hlrnk0fzv56s34qc-harfbuzz-1.8.8-dev/include -I/nix/store/x9snp5bdpf6s4b4wfjdsi1ydwkdgm16s-libjpeg-turbo-1.5.3-dev/include -I/nix/store/yc9pb5rp7638vhcx7807rahhax6vmxqy-libpng-apng-1.6.34-dev/include -I/nix/store/0yz7i5bz64phnhp6fbvi8ljirmy4k31m-libtiff-4.0.9-dev/include -I/nix/store/6lvs7hld8mayamkjadifgswrlb8fqq9f-xz-5.2.4-dev/include -I/nix/store/hn0wq9hff88bml4v3c8fyskw1zhclrlr-pcre2-10.31-dev/include -I/nix/store/yr3d3hgd81i01zmynznjd5vpf95cxq1p-apple-lib-xpc/include -I/nix/store/lnmq68fyjgzp0yxfifxx5k5n35h4nc3x-objc4-osx-10.11.6/include -I/nix/store/567l21pfpvjb3lr5c6hgshasn589raal-libiconv-osx-10.11.6/include -I/nix/store/qjmqrzk8nhn5maa093fhl3finczz72lw-libc++-5.0.2/include -I/nix/store/jd065p28z3ar0q6nljz479f9mm35v5qn-libc++abi-5.0.2/include -I/nix/store/llafcziny4071gkahynh9348yqfwvych-compiler-rt-5.0.2-dev/include -I/nix/store/wnvqlik2lb4i8wfh5z5zs64wady9h1im-qtmultimedia-5.11.1-dev/include -I/nix/store/9nprgv6j52mkc0p5brk5sq670xsgdy5w-qtdeclarative-5.11.1-dev/include -I/nix/store/2sawwhzy4gfmczgfzkallh72fh7k45wk-qtsvg-5.11.1-dev/include -I/nix/store/wkpkjczq0yv0mg258savivx4qc1vakr2-qtxmlpatterns-5.11.1-dev/include -I/nix/store/glngb6whv7q5q8sqxbph9nwf1hacvszk-qtlocation-5.11.1-dev/include -I/nix/store/njijv6rc9xfkga5kdpb2pw2icy0mavm1-qtquickcontrols2-5.11.1-dev/include -I/nix/store/y0kmp0z4xzk62ckrnac3kkmd5jac6zzz-qtwebchannel-5.11.1-dev/include -I/nix/store/1ykgl9bpg9jssb00gzdz431iqh8kcqhi-libwebp-1.0.0/include -I/nix/store/rb42g20l9sdal26dkzvg3axcjv28avfz-libsrtp-2.2.0/include -I/nix/store/8mxm2lgj41vqhfw14rm4j4pfh5ld0bwi-libvpx-1.7.0-dev/include -I/nix/store/dl154cn3s1gzszc8px9r1pwpxza0sdn7-libopus-1.2.1-dev/include -I/nix/store/cklnzvc6pyahr7r0i7nqfrxlzblav31h-libevent-2.1.8-dev/include -I/nix/store/8vhmwix6cgsh7j3kpa02zsqlp25h639w-cctools-port-895-dev/include -I/nix/store/ffxvrp61kssxx1crmbqp0wy5bcy4f8lk-openbsm-1.1/include -I/nix/store/zqbha0b7g3lcwj2m5b02kaisml9lazvq-libunwind-osx-10.11.6/include -I/nix/store/w79n5n1sxa1dzwqj466kv0kbn76g552s-monero-0.12.3.0/include -I/nix/store/a4a7ll8rj273c2al7fay9lm9hyypkj2d-unbound-1.7.3/include -I/nix/store/7c935ncqq268vsi2i0xn4l9j1k8i76fg-readline-6.3p08-dev/include -I/nix/store/zgcay8j39936i2ysgz9sn9c781m5fis1-ncurses-6.1-dev/include -I/nix/store/d5rncndz8fc7wlq53kcnds7gbphmbpc4-boost-1.59.0-dev/include -I/nix/store/p4cg75k1k5x6xaqwbcy6llk6c6fffqs4-pcsclite-1.8.23-dev/include -I/nix/store/pcq24i76i2slbh3xiga465a8bha1zhqj-zeromq-4.2.5/include -I/nix/store/xkapk51lbmqsg2hn1ppi5dp4s5bpw2x9-cppzmq-4.2.3/include -I/nix/store/qjmqrzk8nhn5maa093fhl3finczz72lw-libc++-5.0.2/include/c++/v1 -I/nix/store/2gdwhdzhy4iwkp7fh8v6gy6nxj1zi9pv-clang-wrapper-5.0.2/resource-root/include -I/nix/store/7zcsy4dhx1zcibmqjjnghcmd296bp4ii-Libsystem-osx-10.11.6/include src/daemon/DaemonManager.h -o moc_DaemonManager.cpp
clang++ -c -pipe -stdlib=libc++ -fPIC -fstack-protector -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Wformat -Wformat-security -fstack-protector -fstack-protector-strong -O2 -std=gnu++11 -pthread -Wall -W -fPIC -D_THREAD_SAFE -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/monero/include -Isrc/libwalletqt -Isrc/QR-Code-generator -Isrc -I/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/monero/src -isystem /nix/store/p4cg75k1k5x6xaqwbcy6llk6c6fffqs4-pcsclite-1.8.23-dev/include/PCSC -isystem /nix/store/9nprgv6j52mkc0p5brk5sq670xsgdy5w-qtdeclarative-5.11.1-dev/include/QtQuick -isystem /nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtWidgets -isystem /nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtGui -isystem /nix/store/9nprgv6j52mkc0p5brk5sq670xsgdy5w-qtdeclarative-5.11.1-dev/include/QtQml -isystem /nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtNetwork -isystem /nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/include/QtCore -I. -I/nix/store/5wgzqjpki4xnn9vw1nvfshkpbdlddmwi-apple-framework-OpenGL/Library/Frameworks/OpenGL.framework/Headers -I/nix/store/h9n23fgzmxbi0337n0b9zzh8cbsgh1fv-apple-framework-AGL/Library/Frameworks/AGL.framework/Headers -I/nix/store/hj3nr3ab7xss7g0a8nyj4f8yysy7250d-qtbase-5.11.1-dev/mkspecs/macx-clang -o moc_DaemonManager.o moc_DaemonManager.cpp
clang++ -stdlib=libc++ -fstack-protector -pie -o release/bin/monero-wallet-gui.app/Contents/MacOS/monero-wallet-gui main.o filter.o clipboardAdapter.o oscursor.o WalletManager.o Wallet.o PendingTransaction.o TransactionHistory.o TransactionInfo.o QRCodeImageProvider.o oshelper.o TranslationManager.o TransactionHistoryModel.o TransactionHistorySortFilterModel.o BitBuffer.o QrCode.o QrSegment.o AddressBookModel.o AddressBook.o SubaddressModel.o Subaddress.o zxcvbn.o UnsignedTransaction.o Logger.o MainApp.o DaemonManager.o qrc_translations.o qrc_qml_qmlcache.o RightPanel_qml.o LeftPanel_qml.o version_js.o MiddlePanel_qml.o main_qml.o wizard_WizardConfigure_qml.o wizard_WizardDaemonSettings_qml.o wizard_WizardCreateViewOnlyWallet_qml.o wizard_WizardPassword_qml.o wizard_WizardPasswordInput_qml.o wizard_WizardMain_qml.o wizard_WizardRecoveryWallet_qml.o wizard_WizardOptions_qml.o wizard_WizardCreateWalletFromDevice_qml.o wizard_WizardManageWalletUI_qml.o wizard_WizardMemoTextInput_qml.o wizard_utils_js.o wizard_WizardWelcome_qml.o wizard_WizardCreateWallet_qml.o wizard_WizardFinish_qml.o wizard_WizardDonation_qml.o wizard_WizardPasswordUI_qml.o tabs_Twitter_qml.o tabs_tweetSearch_js.o tabs_TweetsModel_qml.o components_TipItem_qml.o components_PrivacyLevelSmall_qml.o components_DaemonConsole_qml.o components_RadioButton_qml.o components_InlineButton_qml.o components_DaemonManagerDialog_qml.o components_TickDelegate_qml.o components_TitleBar_qml.o components_NetworkStatusItem_qml.o components_LabelButton_qml.o components_MobileHeader_qml.o components_NewPasswordDialog_qml.o components_PasswordDialog_qml.o components_Scroll_qml.o components_TableDropdown_qml.o components_CheckBox_qml.o components_StandardDialog_qml.o components_PrivacyLevel_qml.o components_Style_qml.o components_LabelSubheader_qml.o components_TableHeader_qml.o components_InputDialog_qml.o components_IconButton_qml.o components_LineEdit_qml.o components_DashboardTable_qml.o components_HistoryTableInnerColumn_qml.o components_Label_qml.o components_DatePicker_qml.o components_StandardDropdown_qml.o components_HistoryTable_qml.o components_SearchInput_qml.o components_InputMulti_qml.o components_ProcessingSplash_qml.o components_RemoteNodeEdit_qml.o components_TextBlock_qml.o components_StandardButton_qml.o components_Notifier_qml.o components_AddressBookTable_qml.o components_HistoryTableMobile_qml.o components_ProgressBar_qml.o components_MenuButton_qml.o components_LineEditMulti_qml.o components_CheckBox2_qml.o components_QRCodeScanner_qml.o components_Input_qml.o js_Windows_js.o js_Utils_js.o js_TxUtils_js.o pages_Transfer_qml.o pages_Mining_qml.o pages_Sign_qml.o pages_AddressBook_qml.o pages_Dashboard_qml.o pages_Settings_qml.o pages_Keys_qml.o pages_History_qml.o pages_SharedRingDB_qml.o pages_Receive_qml.o pages_TxKey_qml.o qmlcache_loader.o moc_filter.o moc_clipboardAdapter.o moc_oscursor.o moc_WalletManager.o moc_Wallet.o moc_PendingTransaction.o moc_TransactionHistory.o moc_TransactionInfo.o moc_Transfer.o moc_NetworkType.o moc_oshelper.o moc_TranslationManager.o moc_TransactionHistoryModel.o moc_TransactionHistorySortFilterModel.o moc_AddressBookModel.o moc_AddressBook.o moc_SubaddressModel.o moc_Subaddress.o moc_UnsignedTransaction.o moc_MainApp.o moc_DaemonManager.o   -L/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/monero/lib -lwallet_merged -llmdb -lepee -lunbound -leasylogging -L/usr/local/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/boost/lib -lboost_serialization -lboost_thread-mt -lboost_system -lboost_date_time -lboost_filesystem -lboost_regex -lboost_chrono -lboost_program_options -lssl -lcrypto -ldl -framework PCSC -L/nix/store/v076ylc20f3kq4xvvahamsazgswnz1d1-pcsclite-1.8.23/lib -lpcsclite -lQt5Quick -L/lib -framework DiskArbitration -framework IOKit -lQt5Widgets -lQt5Gui -lQt5Qml -lQt5Network -lQt5Core -framework OpenGL -framework AGL
clang-5.0: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
ld: warning: directory not found for option '-L/private/tmp/nix-build-monero-gui-0.12.3.0.drv-0/source/monero/lib'
ld: library not found for -lboost_thread-mt
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:498: release/bin/monero-wallet-gui.app/Contents/MacOS/monero-wallet-gui] Error 1
builder for '/nix/store/6p81qnr2jvnyk7wn2jr72w4b03dpz9ww-monero-gui-0.12.3.0.drv' failed with exit code 2
error: build of '/nix/store/6p81qnr2jvnyk7wn2jr72w4b03dpz9ww-monero-gui-0.12.3.0.drv' failed
```
</p></details>

--- a/pkgs/applications/altcoins/monero-gui/default.nix
+++ b/pkgs/applications/altcoins/monero-gui/default.nix
@@ -12,13 +12,13 @@ with stdenv.lib;
 
 stdenv.mkDerivation rec {
   name = "monero-gui-${version}";
-  version = "0.12.0.0";
+  version = "0.12.3.0";
 
   src = fetchFromGitHub {
     owner  = "monero-project";
     repo   = "monero-gui";
     rev    = "v${version}";
-    sha256 = "1mg5ival8a2wdp14yib4wzqax4xyvd40zjy9anhszljds1439jhl";
+    sha256 = "1ry0455cgirkc6n46qnlv5p49axjllil78xmx6469nbp3a2r3z7i";
   };
 
   nativeBuildInputs = [ qmake pkgconfig ];
@@ -70,7 +70,8 @@ stdenv.mkDerivation rec {
     cp ${desktopItem}/share/applications/* $out/share/applications
 
     # install translations
-    cp -r release/bin/translations $out/share/
+    mkdir -p $out/share/translations
+    cp translations/*.qm $out/share/translations/
 
     # install icons
     for n in 16 24 32 48 64 96 128 256; do

--- a/pkgs/applications/altcoins/monero-gui/move-log-file.patch
+++ b/pkgs/applications/altcoins/monero-gui/move-log-file.patch
@@ -1,38 +1,27 @@
 diff --git a/main.cpp b/main.cpp
-index c03b160..a8ea263 100644
+index 79223c0..e80b317 100644
 --- a/main.cpp
 +++ b/main.cpp
-@@ -80,14 +80,16 @@ int main(int argc, char *argv[])
- //    qDebug() << "High DPI auto scaling - enabled";
- //#endif
- 
--    // Log settings
--    Monero::Wallet::init(argv[0], "monero-wallet-gui");
--//    qInstallMessageHandler(messageHandler);
--
-     MainApp app(argc, argv);
- 
-     qDebug() << "app startd";
- 
-+    // Log settings
-+    QString logfile =
-+      QStandardPaths::writableLocation(QStandardPaths::CacheLocation)
-+      + "/monero-wallet-gui.log";
-+    Monero::Wallet::init(argv[0], logfile.toUtf8().constData());
-+
-     app.setApplicationName("monero-core");
-     app.setOrganizationDomain("getmonero.org");
-     app.setOrganizationName("monero-project");
-diff --git a/src/libwalletqt/Wallet.cpp b/src/libwalletqt/Wallet.cpp
-index 74649ce..fe1efc6 100644
---- a/src/libwalletqt/Wallet.cpp
-+++ b/src/libwalletqt/Wallet.cpp
-@@ -729,7 +729,7 @@ QString Wallet::getWalletLogPath() const
- #ifdef Q_OS_MACOS
-     return QStandardPaths::standardLocations(QStandardPaths::HomeLocation).at(0) + "/Library/Logs/" + filename;
- #else
--    return QCoreApplication::applicationDirPath() + "/" + filename;
-+    return QStandardPaths::writableLocation(QStandardPaths::CacheLocation) + filename;
+@@ -115,6 +115,9 @@ int main(int argc, char *argv[])
+     QCommandLineOption logPathOption(QStringList() << "l" << "log-file",
+         QCoreApplication::translate("main", "Log to specified file"),
+         QCoreApplication::translate("main", "file"));
++    logPathOption.setDefaultValue(
++        QStandardPaths::writableLocation(QStandardPaths::CacheLocation)
++        + "/monero-wallet-gui.log");
+     parser.addOption(logPathOption);
+     parser.addHelpOption();
+     parser.process(app);
+diff --git a/Logger.cpp b/Logger.cpp
+index 660bafc..dae24d4 100644
+--- a/Logger.cpp
++++ b/Logger.cpp
+@@ -15,7 +15,7 @@ static const QString default_name = "monero-wallet-gui.log";
+ #elif defined(Q_OS_MAC)
+     static const QString osPath = QStandardPaths::standardLocations(QStandardPaths::HomeLocation).at(0) + "/Library/Logs";
+ #else // linux + bsd
+-    static const QString osPath = QStandardPaths::standardLocations(QStandardPaths::HomeLocation).at(0);
++    static const QString osPath = QStandardPaths::standardLocations(QStandardPaths::CacheLocation).at(0);
  #endif
- }
+ 
  

--- a/pkgs/applications/altcoins/monero-gui/move-translations-dir.patch
+++ b/pkgs/applications/altcoins/monero-gui/move-translations-dir.patch
@@ -1,14 +1,13 @@
 diff --git a/TranslationManager.cpp b/TranslationManager.cpp
-index fa39d35..5a410f7 100644
+index e7fc52a..83534cc 100644
 --- a/TranslationManager.cpp
 +++ b/TranslationManager.cpp
-@@ -29,7 +29,7 @@ bool TranslationManager::setLanguage(const QString &language)
- #ifdef Q_OS_MACX
-     QString dir = qApp->applicationDirPath() + "/../Resources/translations";
- #else
+@@ -25,7 +25,7 @@ bool TranslationManager::setLanguage(const QString &language)
+         return true;
+     }
+ 
 -    QString dir = qApp->applicationDirPath() + "/translations";
 +    QString dir = qApp->applicationDirPath() + "/../share/translations";
- #endif
- 
      QString filename = "monero-core_" + language;
-
+ 
+     qDebug("%s: loading translation file '%s' from '%s'",

--- a/pkgs/applications/altcoins/monero/default.nix
+++ b/pkgs/applications/altcoins/monero/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, fetchpatch
+{ stdenv, fetchgit
 , cmake, pkgconfig, git
 , boost, miniupnpc, openssl, unbound, cppzmq
 , zeromq, pcsclite, readline
@@ -11,24 +11,15 @@ with stdenv.lib;
 
 stdenv.mkDerivation rec {
   name    = "monero-${version}";
-  version = "0.12.0.0";
+  version = "0.12.3.0";
 
-  src = fetchFromGitHub {
-    owner  = "monero-project";
-    repo   = "monero";
+  src = fetchgit {
+    url    = "https://github.com/monero-project/monero.git";
     rev    = "v${version}";
-    sha256 = "1lc9mkrl1m8mdbvj88y8y5rv44vinxf7dyv221ndmw5c5gs5zfgk";
+    sha256 = "1609k1qn9xx37a92ai36rajds9cmdjlkqyka95hks5xjr3l5ca8i";
   };
 
   nativeBuildInputs = [ cmake pkgconfig git ];
-
-  patches = [
-    # fix daemon crash, remove with 0.12.1.0 update
-    (fetchpatch {
-      url    = "https://github.com/monero-project/monero/commit/08343ab.diff";
-      sha256 = "0f1snrl2mk2czwk1ysympzr8ismjx39fcqgy13276vcmw0cfqi83";
-    })
-  ];
 
   buildInputs = [
     boost miniupnpc openssl unbound
@@ -39,7 +30,7 @@ stdenv.mkDerivation rec {
     "-DCMAKE_BUILD_TYPE=Release"
     "-DBUILD_GUI_DEPS=ON"
     "-DReadline_ROOT_DIR=${readline.dev}"
-  ];
+  ] ++ optional stdenv.isDarwin "-DBoost_USE_MULTITHREADED=OFF";
 
   hardeningDisable = [ "fortify" ];
 

